lib/statsd/instrument.rb

```diff
@@ -88,6 +100,7 @@ def statsd_distribution(method, name, sample_rate: nil, tags: nil, no_prefix: fa
         define_method(method) do |*args, &block|
           client ||= StatsD.singleton_client
           key = StatsD::Instrument.generate_metric_name(name, self, *args)
+          tags = StatsD::Instrument.generate_tags(tags, self, *args)
```

Comment:
Would be nice if that feature didn't incur an extra cost for users that don't pass a proc.

That said, there's lots of improvement that could be made to these decorators, so maybe we should just do a perf pass  afterwards.

Reply:
I used the following class to run the benchmark with a modified version of send-metrics-to-local-udp-receiver

```
module ActiveMerchant
  class Dummy
    extend StatsD::Instrument

    def ssl_post(arg)
      "OK"
    end
+    statsd_measure(:ssl_post, "metric.name", tags: lambda { |object, args| ["key:#{args.first}"] })  #  branch
-    statsd_measure(:ssl_post, "metric.name", tags: ["key:foo"] )  # master

    def post_with_block(&block)
      block.call if block_given?
    end
  end
end
```

```
Comparison:
UDP batched (branch: dynamic_tags, sha: d650250):   264181.0 i/s
UDP batched (branch: master, sha: e69ab35):   256938.6 i/s - same-ish: difference falls within error
```